parla/task_runtime.py

```diff
@@ -10,19 +11,18 @@
 from itertools import combinations
 from typing import Optional, Collection, Union, Dict, List, Any, Tuple, FrozenSet, Iterable, TypeVar, Deque
 
+# Parla imports
 from parla.device import get_all_devices, Device
 from parla.environments import TaskEnvironmentRegistry, TaskEnvironment
-
 from parla.cpu_impl import cpu
+from parla.cuda import gpu
```

Comment:
I missed this discussion. What was plan? Just allow to import gpu  (with assumption that our targets must have gpu tasks?)

If it is, could you please add TODO or NOTE that it can cause performance overhead?

Reply:
unfortunately, there was no clear solution. our solution for parray and event was just to hide importing if we don't need. if we don't have clear answer, plz just leave TODO to revisit this.

FYI, you can see performance differences through quicksort. 
``` python examples/quicksort.py```

Reply:
I've fixed this. The fix is simple. When checking for GPU architecture, instead of

`if architecture == gpu`

just do

`if architecture.id == 'gpu'`

This requires no import.